dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/repositories/DataRequestRepository.kt

```diff
@@ -0,0 +1,55 @@
+package org.dataland.datalandcommunitymanager.repositories
+
+import org.dataland.datalandbackend.openApiClient.model.DataTypeEnum
+import org.dataland.datalandcommunitymanager.entities.DataRequestEntity
+import org.dataland.datalandcommunitymanager.model.dataRequest.AggregatedDataRequest
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.query.Param
+
+/**
+ * A JPA repository for storing and retrieving data requests
+ */
+
+interface DataRequestRepository : JpaRepository<DataRequestEntity, String> {
+    /** This method gets all data request that are stored for a specific userId.
+     * @param userId defines for which user the data request shall be retrieved
+     * @returns a list with all data request for that user
+     */
+    fun findByUserId(userId: String): List<DataRequestEntity>
+
+    /** This method checks if a data request with the provided params already exists in the database.
+     * @param userId to check for
+     * @param dataRequestCompanyIdentifierValue to check for
+     * @param dataType to check for
+     * @returns a Boolean stating the result of the check
+     */
+    fun existsByUserIdAndDataRequestCompanyIdentifierValueAndDataType(
```

Comment:
What happens if it already exits?

Reply:
You can check in the DataRequestManager.

If t already exists, it will not be created a second time.

So if you (Stephan, userId xyz) have already created a request vor the ISIN 1234 and the framework "p2p", and you request it a second time, it will be ignored the second time.

However the response will not tell you that it is a duplicate. The response looks like everything went as normal.

Reply:
Thank you! Now i understand it. Closed